psl/diagnostics/src/warning.rs

```diff
@@ -25,6 +25,11 @@ impl DatamodelWarning {
         Self::new(message, span)
     }
 
+    pub fn new_missing_index_on_emulated_relation(span: Span) -> DatamodelWarning {
+        let message = "The fields referenced in a `@relation` should be indexed to prevent potential performance penalties. Learn more at https://pris.ly/d/relation-mode-prisma-indexes.".to_owned();
```

Comment:
I think we should revisit the error message: "The fields referenced in a `@relation` should be indexed to prevent potential performance penalties." This will be a default error message for everyone switching from a normal schema to using `relationMode=prisma`, and when writing a schema in relation mode `prisma`.

What do we _really_ want to say here?

Reply:
- we could prefix the message with "When using `relationMode = "prisma"`, "
- uses should know that without foreign keys under the hood, no one (neither prisma or the database) creates indexes on referencing fields mentioned in a `@relation` attribute. This may have performance drawbacks and incur in more full-table scans than expected (e.g., implying highest bills on PlanetScale)

Reply:
I tried something:
```suggestion
        let message = "With `relationMode="prisma", foreign keys will not be created. `@relation` fields will not benefit from being indexed automatically by the database via the foreign key. Which will lead to poor performance when doing queries. We recommend adding an index manually. Learn more at https://pris.ly/d/relation-mode-prisma#indexes.".to_owned();
```
Note that using "reference" might be confusing, as we have a `references` property on `@relation`.

Reply:
I like that. Some iteration:
```suggestion
        let message = "With `relationMode="prisma", no foreign keys are used in your database. `@relation` fields will not benefit from being indexed automatically. This can lead to poor performance querying these fields. We recommend adding an index manually. Learn more at [https://pris.ly/d/relation-mode-prisma#indexes](https://pris.ly/d/relation-mode#indexes)".to_owned();
```

Reply:
Pinging @keerlu for ideas how to communicate this clearly. (also @Druue as she undersand the problem as well)

Reply:
Looks good

Reply:
I'm not entirely sure how to suggest a change but here's my initial thoughts ::
"""
When setting `relationMode` to `"prisma"`, foreign keys will not be used in your database. As a result, `@relation` fields will not benefit from implicit indexing. This can lead to poor performance when querying these fields, and so to improve this, we recommend adding an index manually.
"""

Reply:
Re-elaborating the nice suggestions above, I'd go with:

"""
SQL databases usually silently define an index whenever a foreign key links two tables together. However, with `relationMode = "prisma"`, no foreign keys are used, so `@relation`'s fields will not benefit from implicit indexing.
This can lead to poor performance when querying these fields, we thus recommend adding an index manually.
Learn more at https://pris.ly/d/relation-mode-prisma#indexes
"""

Reply:
I would not start an error message with an explanation of what generally happens, but what the problem is and what lets people build context why they are seeing the error message. "thus" is also a word that some people might not understand, 2 separate sentences are simpler to parse.

Reply:
Ok, how about:

"""
With `relationMode = "prisma"`, no foreign keys are used, so `@relation`'s fields will not benefit from the index usually created by the relational database under the hood.
This can lead to poor performance when querying these fields. We recommend adding an index manually.
Learn more at https://pris.ly/d/relation-mode-prisma#indexes
"""

Reply:
@tomhoule It'd be great if you could give this a final round of review. I'd then include the warning message above and merge this.

Reply:
I could use `indoc!` for the message if needed. Something like:

```rust
pub fn new_missing_index_on_emulated_relation(span: Span) -> DatamodelWarning {
    let message = indoc! {
        r#"With `relationMode = \"prisma\"`, no foreign keys are used, so `@relation`'s fields will not benefit from the index usually created by the relational database under the hood.
        This can lead to poor performance when querying these fields. We recommend adding an index manually.
        Learn more at https://pris.ly/d/relation-mode-prisma#indexes"#
    };
    Self::new(message, span)
}
```

Reply:
`... so `@relation`'s field ...` has an `'s` too much.

Reply:
the `'s` should be there, no? to indicate the possessive of `@relation` @janpio

Reply:
Then it would need to be "a/the @relation's fields".
We call the whole thing "relation field", and `@relation` is the "relation attribute".
So technically, we should probably refine a bit further even.

Reply:
The 'will benefit' part reads oddly to me. I'd suggest something like:

```
"With `relationMode="prisma", no foreign keys are used in your database, and so `@relation` fields will not be indexed automatically...
```

Otherwise looks good to me

Reply:
In terms of referring to the relation, "`@relation`'s field" definitely reads very oddly. "The `@relation`'s fields" is better but not great. Would it be an option to simply refer to "relation fields", rather than "`@relation` fields"?

Reply:
@keerlu I like your wording of 

"""
With `relationMode="prisma", no foreign keys are used in your database, and so `@relation` fields will not be indexed automatically...
"""

but this can also be misleading: not every relational database creates a backing index on `FOREIGN KEY` fields by default (e.g., Postgres doesn't). That's the reason previous sentence adopted **"usually"** in "SQL databases *usually* silently define an index..."